page-objects/components/Menu.js

```diff
@@ -71,16 +79,21 @@ class Menu {
     return $(".c-post__title");
   }
 
+  get searchResultBoxMessage() {
+    return $(".c-search-result-box__message");
+  }
+
   get mobileMenu() {
     return $(".qa-tablet-header__menu");
   }
 
-  get searchResultBoxMessage() {
-    return $(".c-search-result-box__message");
+  async waitForExistHeader() {
+    if (this.header.isExisting) await this.header.waitForDisplayed();
```

Comment:
for mobile, the header is different and should be taken into the consideration. If you will run the test on mobile, that test will break because it the header will not be displayed.